snapcraft/parts/desktop_file.py

```diff
@@ -116,9 +116,15 @@ def write(self, *, gui_dir: Path, icon_path: Optional[str] = None) -> None:
 
         gui_dir.mkdir(parents=True, exist_ok=True)
 
-        # Rename the desktop file to match the app name. This will help
-        # unity8 associate them (https://launchpad.net/bugs/1659330).
-        target = gui_dir / f"{self._app_name}.desktop"
+        desktop_filename = os.path.basename(self._filename)
+
+        # Stop renaming the desktop file. From snapd 2.66 onwards,
```

Comment:
this comment is very good, but I think we should drop the first sentence - it's awkward and potentially confusing to refer to something the code no longer does